packages/proxy/src/util.ts

```diff
@@ -80,82 +81,63 @@ export class ProxyBadRequestError extends Error {
   }
 }
 
-export function parseFilenameFromUrl(url: string): string | undefined {
+export function parseFileMetadataFromUrl(
+  url: string,
+): { filename: string; contentType?: string; url: URL } | undefined {
   try {
     // Handle empty string
     if (!url || url.trim() === "") {
       return undefined;
     }
 
-    // Handle simple filenames without URLs directly
-    if (!url.includes("/") && !url.includes("://")) {
-      try {
-        return decodeURIComponent(url);
-      } catch (e) {
-        return url;
-      }
-    }
-
-    // For URLs without a protocol, add a dummy one to make URL parsing work
-    const normalizedUrl = url.includes("://") ? url : `http://${url}`;
-
     // Use URL to parse complex URLs rather than string splitting
-    const parsedUrl = new URL(normalizedUrl);
-
-    // Extract the pathname
-    const pathname = parsedUrl.pathname;
-
-    // Special case: If hostname contains a file extension and path is empty or just "/"
-    const hostnameMatch = parsedUrl.hostname.match(
-      /\.(pdf|docx?|xlsx?|pptx?|csv|txt|rtf|json|xml|html?|zip|rar|gz|tar|7z)$/i,
-    );
-    if (hostnameMatch && (!pathname || pathname === "/")) {
-      try {
-        return decodeURIComponent(parsedUrl.hostname);
-      } catch (e) {
-        return parsedUrl.hostname;
-      }
-    }
-
-    // If pathname is empty or just "/", there's no filename
-    if (!pathname || pathname === "/") {
+    let parsedUrl: URL | undefined;
+    try {
+      parsedUrl = new URL(url);
+    } catch (e) {
       return undefined;
     }
 
-    // Get the last segment of the path and remove any query parameters
-    const filename = pathname.split("/").pop();
-
-    // Check if filename exists and remove fragment identifier if present
-    if (filename) {
-      try {
-        return decodeURIComponent(filename.split("#")[0]);
-      } catch (e) {
-        return filename.split("#")[0];
-      }
+    // If the URL is not http(s), file cannot be accessed
+    // If pathname is empty or ends with "/", there's no filename to extract
+    if (parsedUrl.protocol !== "http:" && parsedUrl.protocol !== "https:") {
```

Comment:
I was under the impression that we wanted `s3://` and maybe `gs://` to work?

Reply:
nope, that never worked anyways since we use `fetch` to get the file contents for b64-ing